libs/discord/discord.cpp

```diff
@@ -0,0 +1,212 @@
+
+//#include "qcommon/qcommon.h"
+#include "cgame/cg_local.h"
+#include "windows/miniwindows.h"
+#include <time.h>
+#include "discord_game_sdk.h"
+
+#define CLIENT_ID 882369979406753812
+
+using tDiscordCreate = decltype(&DiscordCreate);
+
+static HMODULE discord_sdk;
+static tDiscordCreate pDiscordCreate = nullptr;
+
+struct Application {
+	IDiscordCore* core;
+	IDiscordUserManager* users;
+	IDiscordActivityManager* activities;
+	DiscordUserId user_id;
+};
+
+
+template< size_t size >
+void Q_strncpyz( char (&dest)[size], const char * src )
+{
+	Q_strncpyz( dest, src, size );
+}
+
+
+
+void OnUserUpdated(void* data)
+{
+	Application* app = (Application*)data;
+	DiscordUser user;
+	app->users->get_current_user(app->users, &user);
+	app->user_id = user.id;
+
+	Com_GGPrint("Discord: Hello {}#{}", user.username, user.discriminator);
+}
+
+void UpdateActivityCallback(void* data, EDiscordResult result)
+{
+	Com_GGPrint("Discord: ActivityCallback: {}", result);
+}
+
+const char* log2str(EDiscordLogLevel level)
+{
+	switch(level)
+	{
+		case DiscordLogLevel_Error: return "ERR ";
+		case DiscordLogLevel_Warn: return "WARN";
+		case DiscordLogLevel_Info: return "INFO";
+		case DiscordLogLevel_Debug: return "DBG ";
+		default: return "?--?";
+	}
+}
+
+void OnLog(void* hook_data, EDiscordLogLevel level, const char* message)
+{
+	Com_GGPrint("Discord: [{}]: {}", log2str(level), message);
+}
+
+static Application app = {};
+static IDiscordUserEvents users_events = {};
+static IDiscordActivityEvents activities_events = {};
+
+
+void InitDiscord()
+{
+	ZoneScoped;
+
+	discord_sdk = LoadLibraryW(L"discord_game_sdk.dll");
```

Comment:
How should I handle this, add an #ifdef in this file with `dlopen` and that fun stuff, or create some `Q_LoadLibrary` (heh) that does dlopen on linux and LoadLibrary on windows?

Reply:
We have platform specific stuff under source/unix and source/windows, it can go in there